qsym/pintool/cfg.cpp

```diff
@@ -0,0 +1,114 @@
+#include "cfg.h"
+
+#include <queue>
+
+namespace qsym {
+
+CFG::CFG(const std::string path)
+  : path_(path) {
+
+  if (path.empty()) {
+    exit(-1);
+  }
+
+  std::ifstream ifs(path_, std::ios::in | std::ios::binary);
+  if (ifs.fail()) {
+    LOG_FATAL("cfg file is not available!\n");
+    exit(-1);
+  }
+
+  bo::dynamic_properties dp(bo::ignore_other_properties);
+  dp.property("node_id", get(&Vertex::name,    graph_));
+  dp.property("label",   get(&Vertex::label,   graph_));
+  dp.property("shape",   get(&Vertex::shape,   graph_));
+  dp.property("comment", get(&Vertex::comment, graph_));
+  dp.property("label",   get(&Edge::label,     graph_));
+
```

Comment:
it will be good to comment the meaning of each property

Reply:
also, if we include unnecessary fields, will they slow down the graph searching?